app/sagas/init.js

```diff
@@ -25,16 +28,30 @@ export const initLocalSettings = function* initLocalSettings() {
 
 const restore = function* restore() {
 	try {
-		const { token, server } = yield all({
-			token: UserPreferences.getStringAsync(RocketChat.TOKEN_KEY),
-			server: UserPreferences.getStringAsync(RocketChat.CURRENT_SERVER)
-		});
+		const server = yield UserPreferences.getStringAsync('currentServer');
+		let userId = server
```

Comment:
You check for the existence of `server` here and applies `null` to `userId`.
After that you check for `server` again and dispatch `appStart`.
Can we check for the server only once?
It'd simplify this logic.

Reply:
Done.